### PR TITLE
[Windows|Linux] Use builtin Python proxy capabilities for Certbot-CI

### PR DESCRIPTION
**Pull request by adferrand**
This PR is a part of the actions necessary to make Certbot-CI work on Windows, in order to execute the integration tests on this platform.

I initially used the fully-fledged HTTP proxy [Traefik](https://docs.traefik.io/) to distribute HTTP challenges among several pytest nodes, and so parallelize the integration tests. Traefik for this purpose is overkill. We just want to redirect the ACME server to a pytest node depending on the `Host` header, and we use here a production-grade HTTP proxy for that.

However it was not a problem on Linux, as soon as you can have Docker, because this instance is deployed through it.

But this becomes a problem for Windows, where Docker is not available everywhere, very compelling on its setup, and limited by the implemented network drivers. See my comments here https://github.com/letsencrypt/pebble/pull/240 for more details.

Hopefully Python ships with everything needed to implement a simple HTTP proxy, with strictly what we need for the parallelization of integration tests.

This PR implements this kind of HTTP proxy, and remove the coupling to Traefik.

This PR has been tested successfully with integration tests on Pebble under Linux for Python 2.x and Python 3.x, and the proxy alone has been also tested successfully on Windows (no integration tests can be run for now on this platform).

### CODE REVIEW

File: certbot-ci/certbot_integration_tests/utils/acme_server.py [hunk 1]
```diff
@@ -52,14 +58,15 @@ def setup_acme_server(acme_server, nodes, proxy=True):
     """
     acme_type = 'pebble' if acme_server == 'pebble' else 'boulder'
     acme_xdist = _construct_acme_xdist(acme_server, nodes)
-    workspace, stop = _construct_workspace(acme_type)
+    workspace, server_cleanup = _construct_workspace(acme_type)
```
Comment (bmw) [inline certbot-ci/certbot_integration_tests/utils/acme_server.py]: I think the body of `_construct_workspace` can be simplified and cleaned up a bit with the removal of traefik.

File: certbot-ci/certbot_integration_tests/utils/acme_server.py [hunk 2]
```diff
@@ -52,14 +58,15 @@ def setup_acme_server(acme_server, nodes, proxy=True):
     """
```
Comment (bmw) [inline certbot-ci/certbot_integration_tests/utils/acme_server.py]: We should update the documentation for `proxy`.

File: certbot-ci/certbot_integration_tests/utils/acme_server.py [hunk 3]
```diff
@@ -23,13 +23,19 @@ class ACMEServer(object):
...
+    def __init__(self, acme_xdist, start, server_cleanup):
+        self._proxy_process = None
+        self._server_cleanup = server_cleanup
         self.acme_xdist = acme_xdist
         self.start = start
-        self.stop = stop
+
+    def stop(self):
+        if self._proxy_process:
+            self._proxy_process.terminate()
```
[the remark this answers is not in the document]
Comment (adferrand) [reply certbot-ci/certbot_integration_tests/utils/acme_server.py]: Indeed !

[the remark this answers is not in the document]
Comment (adferrand) [reply certbot-ci/certbot_integration_tests/utils/acme_server.py]: I added the wait.

File: certbot-ci/certbot_integration_tests/utils/acme_server.py [hunk 4]
```diff
@@ -155,60 +162,16 @@ def _prepare_acme_server(workspace, acme_type, acme_xdist):
         raise
 
 
-def _prepare_traefik_proxy(workspace, acme_xdist):
+def _prepare_traefik_proxy(acme_xdist):
     """Configure and launch Traefik, the HTTP reverse proxy"""
```
[the remark this answers is not in the document]
Comment (adferrand) [reply certbot-ci/certbot_integration_tests/utils/acme_server.py]: Done.

File: certbot-ci/certbot_integration_tests/utils/acme_server.py (hunk 1 above)
Comment (adferrand) [reply certbot-ci/certbot_integration_tests/utils/acme_server.py]: Done.

File: certbot-ci/certbot_integration_tests/utils/acme_server.py (hunk 2 above)
Comment (adferrand) [reply certbot-ci/certbot_integration_tests/utils/acme_server.py]: Done.

**Comment by adferrand** [comment]
> I think the [most recent commit](https://github.com/certbot/certbot/pull/7156/commits/6932c98a9b66797cc53c2a10dba5e268bbe583a9) doesn't belong on this PR and you probably meant to push it to #7157.
> 
> This otherwise LGTM though!

Indeed, fixed !

**Pull request merged**